state/src/applicator/wallet_index.rs

```diff
@@ -80,20 +76,85 @@ impl StateApplicator {
     // | Helpers |
     // -----------
 
-    /// Add an order within a given transaction
-    pub(crate) fn add_local_order_with_tx(
-        &self,
-        mut order: NetworkOrder,
-        tx: &StateTxn<RW>,
-    ) -> Result<()> {
-        // Add to the locally managed orders
-        order.local = true;
-        tx.mark_order_local(&order.id)?;
-
-        // Update the order priority, the order, and its nullifier
-        tx.write_order_priority(&order)?;
-        tx.write_order(&order)?;
-        Ok(tx.update_order_nullifier_set(&order.id, order.public_share_nullifier)?)
+    /// Index the orders of a wallet and update denormalized order state
+    fn index_orders_with_tx(&self, wallet: &Wallet, tx: &StateTxn<RW>) -> Result<()> {
+        // Add network orders for the locally managed wallet
+        self.add_network_orders(wallet, tx)?;
+
+        // Update the order -> wallet mapping
+        let nonzero_orders = wallet.get_nonzero_orders().into_keys().collect_vec();
+        tx.index_orders(&wallet.wallet_id, &nonzero_orders)?;
+
+        // Handle new orders and previous orders that are not cancelled
+        self.handle_new_and_cancelled_orders(wallet, tx)?;
+
+        // Change the state of each active order in the wallet to `Created`
+        // to reflect that validity proofs have not been created yet
+        let wallet_id = wallet.wallet_id;
+        for id in wallet.get_nonzero_orders().into_keys() {
+            let mut md = tx
+                .get_order_metadata(wallet_id, id)?
+                .ok_or(StateApplicatorError::MissingEntry(ERR_NO_METADATA))?;
+            md.state = OrderState::Created;
+            self.update_order_metadata_with_tx(tx, md)?;
+        }
```

Comment:
This means that all existing orders (not new or cancelled) will have their states overwritten to `OrderState::Created` - I could see some confusion on the frontend seeing an order go from `OrderState::Matching` to `OrderState::Created`. Thoughts on adding a state for `OrderState::GeneratingProofs` (or something)? If not let's make sure @sehyunc is aware of this so it's represented well on the FE

Reply:
Created can be displayed as "GeneratingProofs" if we like.